### PR TITLE
(refonte) New Suggest Box

### PR DESCRIPTION
**Pull request by gmougeolle**
A tiny well-scoped PR for a change : this only contains a new feature, the Suggest® Box®.
What's new ?

**Metadata -> JS config**

- New system for injecting JS variables from within _Jinja_ : you can now inject `js-config-*` variables as `<meta>` tags in the `<head>` like this : `<meta name="js-config-reuse-url" content="{{ url_for('reuses.list') }}" />` and get them back in Javascript like this : 
```js
import config from "config";
const reuseUrl = config.values.reuseUrl
```

**Components**

- Suggest box and friends : empty result set, search-input for inputing searches
- More robust loaders (skeletons cards) for datasets and reuses
- New Vue cards for displaying datasets and reuses
- New placeholder component that either displays an image or a cool looking placeholder (it might be the glasses image if you're unlucky)

**Ui-kit**

- Dataset card `.no-border` is now `.compact` for semantics

### CODE REVIEW

File: theme/js/components/search/suggest-results.vue [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+<template>
+  <div class="row p-md results">
+    <div class="p-md col col-md-12">
+      <a
+        class="unstyled row-inline justify-between see-all"
+        :href="datasetUrl"
+        :title="$t('@@Rechercher dans tous les jeux de données')"
+      >
+        <h2>
+          Jeux de données <sup>{{ results?.datasets?.length || 0 }}</sup>
```
Comment (abulte) [inline theme/js/components/search/suggest-results.vue:23]:
```suggestion
          Jeux de données <sup>{{ results.datasets.length || 0 }}</sup>
```

Typo or syntax unknown to me?

Comment (abulte) [reply theme/js/components/search/suggest-results.vue:23]: Ok I did not know this one, nice! Github coloring needs to learn it too :-)

File: theme/js/config.js [hunk 2]
```diff
@@ -159,6 +159,24 @@ export const dataset_max_resources_uncollapsed = _jsonMeta(
...
+// New generic `js-config-*` variables : simply add them to `metadata.html` with a meta name="js-config-something", content="yourValue" and import this file.
+// Everything is exported in the `values` variable as key: value pairs
+const valuesPrefix = "js-config-";
+
+export const values = [
+  ...document.querySelectorAll(`meta[name^="${valuesPrefix}"]`),
+].reduce((acc, el) => {
+  if (!el) return acc;
+
+  //Stripping prefix from name, camelizing it too (kebab-case is delicious but hard to use in Javascript)
```
Comment (abulte) [inline theme/js/config.js:171]: 🥗 🍅 🧅

**Review by abulte — APPROVED** [review]

File: theme/js/config.js (hunk 2 above)
Comment (gmougeolle) [reply theme/js/config.js:171]: Pour GitHub c'est sans oignons apparemment

Comment (abulte) [reply theme/js/config.js:171]: Pour moi aussi en vrai, mais je voulais aller au bout de la blague. Je ne recule devant rien.

**Pull request merged**